xbmc/filesystem/BlurayDirectory.cpp

```diff
@@ -7,279 +7,1705 @@
+}
+
+std::map<unsigned int, ClipInformation> lcl_clips;
+
+static constexpr unsigned int OFFSET_STREAM_TYPE = 1;
+static constexpr unsigned int OFFSET_PLAYITEM_PACKET_IDENTIFIER = 2;
+static constexpr unsigned int OFFSET_SUBPATH_SUBPATH_ID = 2;
+static constexpr unsigned int OFFSET_SUBPATH_SUBCLIP_ID = 3;
+static constexpr unsigned int OFFSET_SUBPATH_PACKET_IDENTIFIER = 4;
+static constexpr unsigned int OFFSET_DV_PIP_SUBPATH_ID = 2;
+static constexpr unsigned int OFFSET_DV_PIP_PACKET_IDENTIFIER = 3;
+static constexpr unsigned int OFFSET_STREAM_CODING = 1;
+static constexpr unsigned int OFFSET_STREAM_VIDEO_FLAG_1 = 2;
+static constexpr unsigned int OFFSET_STREAM_VIDEO_FLAG_2 = 3;
+static constexpr unsigned int OFFSET_STREAM_VIDEO_FLAG_3 = 4;
+static constexpr unsigned int OFFSET_STREAM_AUDIO_FLAG_1 = 2;
+static constexpr unsigned int OFFSET_STREAM_AUDIO_LANGUAGE = 3;
+static constexpr unsigned int OFFSET_STREAM_SUB_LANGUAGE = 2;
+static constexpr unsigned int OFFSET_STREAM_TEXT_ENCODING = 2;
+static constexpr unsigned int OFFSET_STREAM_TEXT_LANGUAGE = 3;
+static constexpr unsigned int OFFSET_STREAM_SECONDARY_REFERENCES = 2;
+
+StreamInformation ParseStream(std::vector<char>& buffer,
+                              unsigned int& offset,
+                              STREAM_TYPE streamType)
+{
+  StreamInformation streamInformation;
+  unsigned int length{CUtil::GetByte(buffer, offset)};
+
+  const BLURAY_STREAM_TYPE type{CUtil::GetByte(buffer, offset + OFFSET_STREAM_TYPE)};
+  streamInformation.type = type;
+  switch (type)
   {
-    bd_close(m_bd);
-    m_bd = nullptr;
+    case BLURAY_STREAM_TYPE::PLAYITEM: // a stream of the clip used by the PlayItem
+    {
+      streamInformation.packetIdentifier =
+          CUtil::GetWord(buffer, offset + OFFSET_PLAYITEM_PACKET_IDENTIFIER);
+      break;
+    }
+    case BLURAY_STREAM_TYPE::
+        SUBPATH: // a stream of the clip used by a SubPath (types 2, 3, 4, 5, 6, 8 or 9)
+    {
+      streamInformation.subpathId = CUtil::GetWord(buffer, offset + OFFSET_SUBPATH_SUBPATH_ID);
+      streamInformation.subclipId = CUtil::GetWord(buffer, offset + OFFSET_SUBPATH_SUBCLIP_ID);
+      streamInformation.packetIdentifier =
+          CUtil::GetWord(buffer, offset + OFFSET_SUBPATH_PACKET_IDENTIFIER);
+      break;
+    }
+    case BLURAY_STREAM_TYPE::
+        SUBPATH_INMUX_SYNCHRONOUS_PIP: // a stream of the clip used by a SubPath (type 7)
+    case BLURAY_STREAM_TYPE::
+        SUBPATH_DOLBY_VISION_LAYER: // a stream of the clip used by a SubPath (type 10)
+    {
+      streamInformation.subpathId = CUtil::GetWord(buffer, offset + OFFSET_DV_PIP_SUBPATH_ID);
+      streamInformation.packetIdentifier =
+          CUtil::GetWord(buffer, offset + OFFSET_DV_PIP_PACKET_IDENTIFIER);
+      break;
+    }
+  }
+  offset += length + 1;
+
+  length = CUtil::GetByte(buffer, offset);
+  const unsigned int coding{CUtil::GetByte(buffer, offset + OFFSET_STREAM_CODING)};
```

Comment:
would be better to use a proper fixed-width type depending on what content you are dealing with (e.g. `std::uint8_t` or `std::byte` here)